src/devices/Pigpio/PigpiodIf.cs

```diff
@@ -0,0 +1,2789 @@
+		{ return pigpio_command(PI_CMD_SLRC, (int)user_gpio, 0); }
+
+		public int bb_serial_invert(UInt32 user_gpio, UInt32 invert)
+		{ return pigpio_command(PI_CMD_SLRI, (int)user_gpio, (int)invert); }
+
+		public int i2c_open(UInt32 i2c_bus, UInt32 i2c_addr, UInt32 i2c_flags)
+		{
+			GpioExtent[] exts = new GpioExtent[] { new GpioExtent() };
+
+			/*
+			p1=i2c_bus
+			p2=i2c_addr
+			p3=4
+			## extension ##
+			uint32_t i2c_flags
+			*/
+
+			exts[0].Contents = UInt32ToBytes(i2c_flags);
+
+			return pigpio_command_ext(PI_CMD_I2CO, (int)i2c_bus, (int)i2c_addr, exts);
+		}
+
+		public int i2c_close(UInt32 handle)
+		{ return pigpio_command(PI_CMD_I2CC, (int)handle, 0); }
+
+		public int i2c_write_quick(UInt32 handle, UInt32 bit)
+		{ return pigpio_command(PI_CMD_I2CWQ, (int)handle, (int)bit); }
+
+		public int i2c_write_byte(UInt32 handle, UInt32 val)
+		{ return pigpio_command(PI_CMD_I2CWS, (int)handle, (int)val); }
+
+		public int i2c_read_byte(UInt32 handle)
+		{ return pigpio_command(PI_CMD_I2CRS, (int)handle, 0); }
+
+		public int i2c_write_byte_data(UInt32 handle, UInt32 reg, UInt32 val)
+		{
+			GpioExtent[] exts = new GpioExtent[] { new GpioExtent() };
+
+			/*
+			p1=handle
+			p2=reg
+			p3=4
+			## extension ##
+			uint32_t val
+			*/
+
+			exts[0].Contents = UInt32ToBytes(val);
+
+			return pigpio_command_ext(PI_CMD_I2CWB, (int)handle, (int)reg, exts);
+		}
+
+		public int i2c_write_word_data(UInt32 handle, UInt32 reg, UInt32 val)
+		{
+			GpioExtent[] exts = new GpioExtent[] { new GpioExtent() };
+
+			/*
+			p1=handle
+			p2=reg
+			p3=4
+			## extension ##
+			uint32_t val
+			*/
+
+			exts[0].Contents = UInt32ToBytes(val);
+
+			return pigpio_command_ext(PI_CMD_I2CWW, (int)handle, (int)reg, exts);
+		}
+
+		public int i2c_read_byte_data(UInt32 handle, UInt32 reg)
+		{ return pigpio_command(PI_CMD_I2CRB, (int)handle, (int)reg); }
+
+		public int i2c_read_word_data(UInt32 handle, UInt32 reg)
+		{ return pigpio_command(PI_CMD_I2CRW, (int)handle, (int)reg); }
+
+		public int i2c_process_call(UInt32 handle, UInt32 reg, UInt32 val)
+		{
+			GpioExtent[] exts = new GpioExtent[] { new GpioExtent() };
+
+			/*
+			p1=handle
+			p2=reg
+			p3=4
+			## extension ##
+			uint32_t val
+			*/
+
+			exts[0].Contents = UInt32ToBytes(val);
+
+			return pigpio_command_ext(PI_CMD_I2CPK, (int)handle, (int)reg, exts);
+		}
+
+		public int i2c_write_block_data(UInt32 handle, UInt32 reg, byte[] buf)
+		{
+			GpioExtent[] exts = new GpioExtent[] { new GpioExtent() };
+
+			/*
+			p1=handle
+			p2=reg
+			p3=count
+			## extension ##
+			char buf[count]
+			*/
+
+			exts[0].Contents = buf;
+
+			return pigpio_command_ext(PI_CMD_I2CWK, (int)handle, (int)reg, exts);
+		}
+
+		public int i2c_read_block_data(UInt32 handle, UInt32 reg, byte[] buf)
+		{
+			int bytes;
+
+			lock (LockObject)
+			{
+				bytes = pigpio_command(PI_CMD_I2CRK, (int)handle, (int)reg);
+
+				if (bytes > 0)
+				{
+					// 32バイト固定になっていたが、不明
```

Comment:
huh?

Reply:
Translates to "It was fixed at 32 bytes, but unknown"